### PR TITLE
Handle contact modern mecanics

### PR DESCRIPTION
**Pull request by Mixone-FinallyHere**
Handles contact with modern mecanics and leaves easy framework to add or remove item effects and abilities that can prevent contact

It includes also the stuff for PR #310 i edited that on main instead of a branch so its all together (sorry)

### CODE REVIEW

**Comment by BluRosie** [comment]
so...  we know that this covers the existing contact things in hg-engine, do we know about everything beyond this as well?  or are we thinking of tackle it as needed?

**Comment by Mixone-FinallyHere** [comment]
> so... we know that this covers the existing contact things in hg-engine, do we know about everything beyond this as well? or are we thinking of tackle it as needed?

Im not SUPER familiar with like the most modern mecanics but between mojo and aero they mentioned all cases for contact that exist (PGloves + PPads and Long Reach)

The idea is that since the door is left open to easily add others, if in the future we see more its easy to simply append to this function I would say

TL;DR: As far as I know this tackles all possible use cases ASSUMING that neutralizing gas will be implemented at CheckAbility level

File: src/battle/other_battle_calculators.c [hunk 1]
```diff
@@ -2734,6 +2734,55 @@ BOOL CheckStrongWindsWeaken(struct BattleSystem *bw, struct BattleStruct *sp) {
...
+        // Kept in case people add their own
+        // || HeldItemHoldEffectGet(sp, sp->defence_client) != OTHER_HOLD_EFFECT_THAT_PREVENTS_DEFENDER_CONTACT
+        ) {
+            return FALSE;
+    }
+
+    // Does the move make contact vanilla
+    if (IsContactBeingMade(bw, sp)) {
+        return TRUE;
+    }
```
Comment (BluRosie) [inline src/battle/other_battle_calculators.c:2781]: where is this linked into vanilla?  currently seems like it would recursively call itself and would likely stack overflow

Comment (BluRosie) [reply src/battle/other_battle_calculators.c:2781]: should probably just be `sp->moveTbl[sp->current_move_index].flag & FLAG_CONTACT`

**Review by BluRosie — CHANGES_REQUESTED** [review]

**Review by BluRosie — APPROVED** [review]
